### PR TITLE
Topic register

### PR DESCRIPTION
**Pull request by urviljoshi**

### CODE REVIEW

File: hub/hub_service.bal [hunk 1]
```diff
@@ -267,24 +261,44 @@ websubhub:Service hubService = service object {
...
+
+        if !topicAvailable {
+            log:printInfo("Topic not found - Auto registering topic", topic = topicName);
+            websubhub:TopicRegistration topicRegistrationMsg = {
+                topic: topicName,
+                hubMode: "register"
+            };
+            websubhub:TopicRegistrationError? registerTopicResult = self.registerTopic(topicRegistrationMsg);
+            if registerTopicResult is websubhub:TopicRegistrationError {
+                log:printError("Auto registering topic - Successfully registered failed with error", err = registerTopicResult.message());
```
Comment (vishwa-vyom) [inline hub/hub_service.bal:292]: In this case, whyt we are not returning any error to publish or subscribe API ?

File: hub/start_hub.bal [hunk 2]
```diff
@@ -31,6 +32,7 @@ public function main() returns error? {
     // Initialize the Hub
     _ = @strand {thread: "any"} start syncRegsisteredTopicsCache();
     _ = @strand {thread: "any"} start syncSubscribersCache();
+
```
Comment (vishwa-vyom) [inline hub/start_hub.bal:36]: Why is this new line and above regex import required ?

**Review by vishwa-vyom — CHANGES_REQUESTED** [review]

File: hub/start_hub.bal (hunk 2 above)
Comment (urviljoshi) [reply hub/start_hub.bal:36]: done

File: hub/hub_service.bal (hunk 1 above)
Comment (urviljoshi) [reply hub/hub_service.bal:292]: there is only one error returned by api registered topic already exist, we will not stop program for that right?

**Review by vishwa-vyom — APPROVED** [review]

**Pull request merged**